modules/cryptoticker/binance/binance.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package binance provides crypto coin ticker using the Binance API
+// available at https://api.binance.com.
+package binance // import "barista.run/modules/cryptoticker/binance"
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"barista.run/modules/cryptoticker"
+)
+
+// binananceCryptoTicker represents an binance ticker json response.
+type binanceCryptoTicker struct {
+	Symbol				string		`json:"symbol"`
+	PriceChange			float32		`json:"priceChange,string"`
+	PriceChangePercent	float32		`json:"priceChangePercent,string"`
+	WeightedAvgPrice	float32		`json:"weightedAvgPrice,string"`
+	PrevClosePrice		float32		`json:"prevClosePrice,string"`
+	LastPrice			float32		`json:"lastPrice,string"`
+	LastQty				float32		`json:"lastQty,string"`
+	BidPrice			float32		`json:"bidPrice,string"`
+	AskPrice			float32		`json:"askPrice,string"`
+	OpenPrice			float32		`json:"openPrice,string"`
+	HighPrice			float32		`json:"highPrice,string"`
+	LowPrice			float32		`json:"lowPrice,string"`
+	Volume				float32		`json:"volume,string"`
+	QuoteVolume			float32		`json:"quoteVolume,string"`
+	OpenTime			uint64		`json:"openTime"`
+	CloseTime			uint64		`json:"closeTime"`
+	FirstID				uint64		`json:"firstId"`
+	LastID				uint64		`json:"lastId"`
+	Count				uint64		`json:"count"`
+}
+
+// Provider wraps a coin symbol so that it can be used as a cryptoticker.Provider.
+type Provider struct{
+	Symbol string
+}
+
+// GetTicker gets ticker information from Binance.
+func (binance Provider) GetTicker() (cryptoticker.CryptoTicker, error) {
+	srvURL := url.URL{
+		Scheme:		"https",
+		Host:		"api.binance.com",
+		Path:		"/api/v3/ticker/24hr",
+		RawQuery:	"symbol=" + binance.Symbol,
+	}
```

Comment:
Looks like you picked the weather module as the base to start from. Which is fine, but it makes this overly generic :)

The weather module was written in a way that allows external providers to be loaded, since good weather data is hard to find. This would let users plug in local sources (even hyper-local sources, such as a personal weather station) without adding significant work.

In the case of this module, I'm not sure the benefits are worth the extra indirection. It seems that you can feed any number of symbols to this API, so it's not the case that users will need to bring their own API if they want to track multiple symbols. (But I'm not familiar with this alternate currency. Maybe each API has a subset of the currencies? Do currencies provide a canonical API?) If you still think this is reasonable, I'm fine with keeping this.